ee/clickhouse/client.py

```diff
@@ -131,16 +134,25 @@ def sync_execute(query, args=None, settings=None, with_column_types=False):
             if app_settings.SHELL_PLUS_PRINT_SQL:
                 print()
                 print(format_sql(query, args))
+
+            sql, tags = _annotate_tagged_query(query, args)
+            timeout_task = QUERY_TIMEOUT_THREAD.schedule(_notify_of_slow_query_failure, tags)
```

Comment:
This is happening inside the CH pool, and depending on load, there can be as many timeout_tasks scheduled at the same time, as we have max CH connections.

Let's say, N scheduled on the `SingleThreadedTimer`, which get added depending on whichever came first.

This implies, `schedule()`, and thus `start()` is called multiple times, which seems wrong? https://docs.python.org/3/library/threading.html#threading.Thread.start - can only be called once on a thread.

Reply:
Nvm, I see the override

Reply:
`start` is overloaded to make it safe to call multiple times. Will add a comment to that effect.

Goal was to only start the thread once first task comes in.